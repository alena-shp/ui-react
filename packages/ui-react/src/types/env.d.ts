/// <reference types="vite/client" />

/* Custom ENV
interface ImportMetaEnv {
    readonly MY_ENV: string
}
*/

interface ImportMeta {
    readonly env: ImportMetaEnv
}
