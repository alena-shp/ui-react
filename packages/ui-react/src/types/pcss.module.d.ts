declare module '*.module.pcss' {
    const classes: Record<string, string>;

    export default classes;
}
