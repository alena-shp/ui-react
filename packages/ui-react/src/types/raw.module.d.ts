declare module '*?raw' {
    const text: string;

    export default text;
}
