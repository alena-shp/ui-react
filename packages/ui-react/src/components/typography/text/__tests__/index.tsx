import React from 'react';
import * as renderer from 'react-test-renderer';

import { Text, typographyTextSize, typographyTextTags } from '..';
import { typographyColor } from '../../types';

describe('Text', () => {
    typographyTextSize.map((size) => {
        test(size, () => {
            const tree = renderer.create(
                <Text
                    key={size}
                    presetSize={size}
                >
                    {size}
                </Text>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});

describe('Text', () => {
    typographyTextTags.map((tag) => {
        test(tag, () => {
            const tree = renderer.create(
                <Text
                    key={tag}
                    tagName={tag}
                >
                    {tag}
                </Text>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});

describe('Text', () => {
    typographyColor.map((color) => {
        test(color, () => {
            const tree = renderer.create(
                <Text
                    key={color}
                    presetColor={color}
                >
                    {color}
                </Text>
            ).toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
