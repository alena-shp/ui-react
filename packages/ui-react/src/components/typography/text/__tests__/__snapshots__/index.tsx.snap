// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text a 1`] = `
<a
  className="typography typography_size-body typography_color-basic"
>
  a
</a>
`;

exports[`Text b 1`] = `
<b
  className="typography typography_size-body typography_color-basic"
>
  b
</b>
`;

exports[`Text basic 1`] = `
<p
  className="typography typography_size-body typography_color-basic"
>
  basic
</p>
`;

exports[`Text body 1`] = `
<p
  className="typography typography_size-body typography_color-basic"
>
  body
</p>
`;

exports[`Text body-bold 1`] = `
<p
  className="typography typography_size-body-bold typography_color-basic"
>
  body-bold
</p>
`;

exports[`Text brand 1`] = `
<p
  className="typography typography_size-body typography_color-brand"
>
  brand
</p>
`;

exports[`Text brand-1 1`] = `
<p
  className="typography typography_size-body typography_color-brand-1"
>
  brand-1
</p>
`;

exports[`Text caption 1`] = `
<p
  className="typography typography_size-caption typography_color-basic"
>
  caption
</p>
`;

exports[`Text caption-bold 1`] = `
<p
  className="typography typography_size-caption-bold typography_color-basic"
>
  caption-bold
</p>
`;

exports[`Text disabled 1`] = `
<p
  className="typography typography_size-body typography_color-disabled"
>
  disabled
</p>
`;

exports[`Text em 1`] = `
<em
  className="typography typography_size-body typography_color-basic"
>
  em
</em>
`;

exports[`Text error 1`] = `
<p
  className="typography typography_size-body typography_color-error"
>
  error
</p>
`;

exports[`Text headers 1`] = `
<p
  className="typography typography_size-body typography_color-headers"
>
  headers
</p>
`;

exports[`Text i 1`] = `
<i
  className="typography typography_size-body typography_color-basic"
>
  i
</i>
`;

exports[`Text inherit 1`] = `
<p
  className="typography typography_size-body typography_color-inherit"
>
  inherit
</p>
`;

exports[`Text inverse 1`] = `
<p
  className="typography typography_size-body typography_color-inverse"
>
  inverse
</p>
`;

exports[`Text large 1`] = `
<p
  className="typography typography_size-large typography_color-basic"
>
  large
</p>
`;

exports[`Text large-bold 1`] = `
<p
  className="typography typography_size-large-bold typography_color-basic"
>
  large-bold
</p>
`;

exports[`Text legend 1`] = `
<legend
  className="typography typography_size-body typography_color-basic"
>
  legend
</legend>
`;

exports[`Text link 1`] = `
<p
  className="typography typography_size-body typography_color-link"
>
  link
</p>
`;

exports[`Text p 1`] = `
<p
  className="typography typography_size-body typography_color-basic"
>
  p
</p>
`;

exports[`Text secondary 1`] = `
<p
  className="typography typography_size-body typography_color-secondary"
>
  secondary
</p>
`;

exports[`Text span 1`] = `
<span
  className="typography typography_size-body typography_color-basic"
>
  span
</span>
`;

exports[`Text strong 1`] = `
<strong
  className="typography typography_size-body typography_color-basic"
>
  strong
</strong>
`;

exports[`Text success 1`] = `
<p
  className="typography typography_size-body typography_color-success"
>
  success
</p>
`;

exports[`Text warning 1`] = `
<p
  className="typography typography_size-body typography_color-warning"
>
  warning
</p>
`;
